### PR TITLE
chore(mongo): case insensitive like search

### PR DESCRIPTION
**Pull request by skjindal93**

### CODE REVIEW

File: document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java [hunk 1]
```diff
@@ -279,7 +279,9 @@ private void parseOrderByQuery(List<OrderBy> orderBys, Map<String, Object> order
           map.put(filter.getFieldName(), value);
           break;
         case LIKE:
-          map.put(filter.getFieldName(), new BasicDBObject("$regex", value));
+          // Case insensitive regex search
```
Comment (skjindal93) [inline document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java:282]: Only logic change. Rest are tests and formatting changes

File: document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java [hunk 2]
```diff
@@ -279,7 +279,9 @@ private void parseOrderByQuery(List<OrderBy> orderBys, Map<String, Object> order
...
+          map.put(
```
Comment (kotharironak) [inline document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java:283]: would this leave something is searched case insensitive in a document store, and but OLAP at case sensitive?  

Does LIKE is generally case-insensitive? Or we kind need both kind of filter `LIKE` and `LIKEI`?

Comment (skjindal93) [reply document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java:283]: OLAP is already case insensitive

**Review by kotharironak — APPROVED** [review]
lgtm

**Pull request merged**